product_multicompany_submit/tests/test_multicompany_submit.py

```diff
@@ -0,0 +1,165 @@
+
+        self.company_approver = self.env["res.company"].create(
+            {"name": "company of approver"}
+        )
+        self.company_submitter = self.env["res.company"].create(
+            {"name": "company of submitter"}
+        )
+        self.company_other = self.env["res.company"].create(
+            {"name": "company of other user"}
+        )
+
+        self.user_approver = self.env["res.users"].create(
+            {
+                "name": "approver",
+                "login": "approver",
+                "company_ids": self.company_approver.ids,
+                "company_id": self.company_approver.id,
+                "groups_id": [
+                    (
+                        6,
+                        0,
+                        [
+                            self.ref(
+                                "product_multicompany_submit."
+                                "group_multicompany_product_approve"
+                            ),
+                            self.ref("base.group_user"),
+                        ],
+                    )
+                ],
+            }
+        )
+
+        self.user_submitter = self.env["res.users"].create(
+            {
+                "name": "submitter",
+                "login": "submitter",
+                "company_ids": self.company_submitter.ids,
+                "company_id": self.company_submitter.id,
+                "groups_id": [
+                    (
+                        6,
+                        0,
+                        [
+                            self.ref(
+                                "product_multicompany_submit."
+                                "group_multicompany_product_submit"
+                            ),
+                            self.ref("base.group_user"),
+                        ],
+                    )
+                ],
+            }
+        )
+
+        self.user_other = self.env["res.users"].create(
+            {
+                "name": "other user",
+                "login": "other user",
+                "company_ids": self.company_other.ids,
+                "company_id": self.company_other.id,
+                "groups_id": [(6, 0, [self.ref("base.group_user")],)],
+            }
+        )
+
+        self.product = self.env["product.template"].create(
+            {"name": "some product", "company_id": self.company_submitter.id}
+        )
+
+    def test_permissions(self):
+        self.assertEqual(
+            self.product.state_multicompany_submit, "not_submitted"
+        )
+        self.assertEqual(
+            self.product.multicompany_origin_company_id, self.company_submitter
+        )
+        self.assertEqual(self.product.company_id, self.company_submitter)
+
+        # NOT SUBMITTED permissions: approver, other user should not have them
+        with self.assertRaises(AccessError):
+            self.env["product.template"].sudo(self.user_approver.id).browse(
```

Comment:
a simplier syntax can be 
self.product.sudo(self.user_approver.id).read(["name"])